hpke/hpke.go

```diff
@@ -0,0 +1,253 @@
+// Package hpke implements the Hybrid Public Key Encryption (HPKE) as specified
+// by draft-irtf-cfrg-hpke-06.
+//
+// HPKE works for any combination of an asymmetric-key encapsulation mechanism
+// (KEM), a key derivation function (KDF), and an authenticated symmetric-key
+// encryption scheme with additional data (AEAD).
+//
+// Specification in
+// https://www.ietf.org/archive/id/draft-irtf-cfrg-hpke-06.html
```

Comment:
I would link to the datatracker page here. You already mentioned that draft-06 is implemented above.